src/test/java/org/breedinginsight/daos/BrAPIDAOUtilUnitTest.java

```diff
@@ -0,0 +1,172 @@
+        testProgram.setBrapiUrl("http://brapiserver:8083");
+
+        //Create Germplasm
+        List<String> shortNames = new ArrayList<String>();
+        shortNames.add("A");
+        shortNames.add("B");
+        shortNames.add("C");
+        shortNames.add("D");
+        shortNames.add("E");
+        shortNames.add("F");
+        shortNames.add("G");
+        shortNames.add("H");
+        shortNames.add("I");
+        shortNames.add("J");
+
+        germplasm = new ArrayList();
+        germplasmMap = new HashMap<>();
+
+        for (int i = 0; i < shortNames.size(); i++) {
+            String entry = String.valueOf(i + 2);
+            String nameIndex = String.valueOf(i+1);
+            BrAPIGermplasm testGermplasm = new BrAPIGermplasm();
+            testGermplasm.setGermplasmName(String.format("Germplasm %1$s [TEST-%2$s]",
+                    shortNames.get(i), nameIndex));
+            testGermplasm.setSeedSource("Wild");
+            testGermplasm.setAccessionNumber(nameIndex);
+            testGermplasm.setDefaultDisplayName(String.format("Germplasm %s", shortNames.get(i)));
+            if(!shortNames.get(i).equals("A")) {
+                testGermplasm.setPedigree("Germplasm A [TEST-1]");
+            }
+            JsonObject additionalInfo = new JsonObject();
+            additionalInfo.addProperty("importEntryNumber", entry);
+            additionalInfo.addProperty("breedingMethod", "Allopolyploid");
+            testGermplasm.setAdditionalInfo(additionalInfo);
+            List<BrAPIExternalReference> externalRef = new ArrayList<>();
+            BrAPIExternalReference testReference = new BrAPIExternalReference();
+            testReference.setReferenceSource(referenceSource);
+            testReference.setReferenceID(UUID.randomUUID().toString());
+            externalRef.add(testReference);
+            testGermplasm.setExternalReferences(externalRef);
+            germplasm.add(testGermplasm);
+            germplasmMap.put(testReference.getReferenceID(), testGermplasm);
+        }
+
+        // Set query params
+        germplasmSearch = new BrAPIGermplasmSearchRequest();
+        germplasmSearch.externalReferenceIDs(List.of(testProgram.getId().toString()));
+        germplasmSearch.externalReferenceSources(List.of(String.format("%s/programs", referenceSource)));
+    }
+
+    @Test
+    @SneakyThrows
+    public void searchGermplasmPost() {
+        // Set query params
+        germplasmSearch.setPage(0);
+
+        List<BrAPIGermplasm> searchResult = brAPIDAOUtil.search(
+                searchBody -> {
+                  return getStubbedGermplasm(searchBody.getPage(), searchBody.getPageSize());
+                },
+                (searchId, page, pageSize) -> {
+                    return null;
+                },
+                germplasmSearch
+        );
+
+        //Check if all requested germplasm are returned
+        for (BrAPIGermplasm germplasm : searchResult) {
```

Comment:
Shouldn't it be iterating through the germplasmMap instead, since it's checking if all the map values are present in the search results?

Reply:
Yep, it should. The unit test is fixed, now.